src/openfermion/ops/_fermion_operator.py

```diff
@@ -99,6 +117,139 @@ def number_operator(n_orbitals, orbital=None, coefficient=1.):
     return operator
 
 
+def sz_operator(n_spatial_orbitals, up_map=up_index, down_map=down_index):
+    """Return the sz operator.
+
+    SZ = 0.5 * \sum_{i = 1}^{n} n_{i \alpha} - n_{i, \beta}
+
+    Args:
+        n_spatial_orbitals: number of spatial orbitals (n_qubits // 2).
+        up_map: function mapping a spatial index to a spin-orbital index.
+                Default is the canonical spin-up corresponds to even
+                spin-orbitals and spin-down corresponds to odd spin-orbitals
+        down_map: function mapping spatial index to spin-orbital index.
+                  Default is canonical spin-up corresponds to even
+                  spin-orbitals and spin-down corresponds to odd
+                  spin-orbitals.
+
+    Returns:
+        operator (FermionOperator): corresponding to the sz operator over
+        n_spatial_orbitals.
+
+    Warnings:
+        Default assumes a number occupation vector representation with even
+        spin-less fermions corresponding to spin-up (alpha) and odd spin-less
+        fermions corresponding to spin-down (beta).
+    """
+    if not isinstance(n_spatial_orbitals, int):
+        raise TypeError("n_orbitals must be specified as an integer")
+
+    operator = FermionOperator()
+    n_spinless_orbitals = 2 * n_spatial_orbitals
+    for ni in range(n_spatial_orbitals):
+        operator += number_operator(n_spinless_orbitals, up_map(ni), 0.5) + \
+                    number_operator(n_spinless_orbitals, down_map(ni), -0.5)
+
+    return operator
+
+
+def s_plus_operator(n_spatial_orbitals, up_map=up_index, down_map=down_index):
+    """Return the s+ operator.
+
+    S+ = \sum_{i=1}^{n} a_{n, \alpha}^{\dagger}a_{n, \beta}
+
+     Args:
+        n_spatial_orbitals: number of spatial orbitals (n_qubits // 2).
```

Comment:
Actually, I think this would be (n_qubits + 1) // 2. Right?

Reply:
Where does the +1 come from?  n_qubits is supposed to stand for the number of qubits which corresponds to the number of spin-orbitals.  Thus number of qubits / 2 is the number of spatial functions.

Reply:
you wrote floor division

Reply:
:sweat_smile: right you are.  :heavy_division_sign: